miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -845,11 +846,24 @@ public ModelAndView editBulkLibraries(@PathVariable String libraryIds, ModelMap
           lai = libraryAdditionalInfoService.get(library.getId());
         }
         libraryDtos.add(Dtos.asDto(library, lai));
+        if (!isDetailedSampleEnabled()) {
+          // Do nothing about sample classes.
+        } else if (sampleClass == null) {
+          sampleClass = ((SampleAdditionalInfo) library.getSample()).getSampleClass();
+        } else if (((SampleAdditionalInfo) library.getSample()).getSampleClass().getId() != sampleClass.getId()) {
+          throw new IOException("Can only update libraries when samples all have the same class.");
```

Comment:
This check should also be performed in the JS on List Libraries so users see an alert rather than a stacktrace.